### DISCUSSION TITLE
c++ deployment error,and how do you deploy c++

### DISCUSSION THREAD

**Discussion by soloplayl** [ultralytics/ultralytics#1275] [Q&A]
cv::dnn::Net net = cv::dnn::readNetFromONNX("/home/soloplayl/ultralytics/yolov8s.onnx")
I get some errors when I use it:

[ERROR:0@0.089] global /home/soloplayl/下载/opencv-4.6.0/modules/dnn/src/onnx/onnx_importer.cpp (1018) handleNode DNN/ONNX: ERROR during processing node with 2 inputs and 1 outputs: [Add]:(onnx_node!/model.22/Add) from domain='ai.onnx'
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.6.0) /home/soloplayl/下载/opencv-4.6.0/modules/dnn/src/onnx/onnx_importer.cpp:1040: error: (-2:Unspecified error) in function 'handleNode'
> Node [Add@ai.onnx]:(onnx_node!/model.22/Add) parse error: OpenCV(4.6.0) /home/soloplayl/下载/opencv-4.6.0/modules/dnn/src/onnx/onnx_importer.cpp:1464: error: (-215:Assertion failed) blob_0.size == blob_1.size in function 'parseBias'
> 

So how do you deploy c++

**Comment by AyushExel** [comment]
Please follow the C++ example here by @JustasBart https://github.com/ultralytics/ultralytics/tree/main/examples/YOLOv8-CPP-Inference

**Reply by soloplayl** [reply]
I used that sample file, but when I read the onnx model, I started to report errors.  Is it related to my OpenCV version?  What is your OpenCV version

**Reply by glenn-jocher** [reply]
@soloplayl the error you're encountering suggests there might be an incompatibility between the ONNX model and the version of OpenCV you're using. It's possible that the ONNX model uses operations or layers that are not supported by OpenCV 4.6.0.

To address this issue, you could try the following steps:
1. Ensure that you are using the latest version of OpenCV, as newer versions tend to have better ONNX support.
2. Check if there are any known issues with ONNX model compatibility in the OpenCV version you're using.
3. Consider using a different method for deploying the model in C++, such as using ONNX Runtime, which is specifically designed to work with ONNX models and might provide better compatibility.

If you continue to experience issues, you might want to open an issue on the OpenCV GitHub repository or seek help from the OpenCV community, as they might have more insights into the compatibility of ONNX models with different versions of OpenCV.

**Comment by JustasBart** [comment]
Hi @soloplayl that to me does seem like it might be an OpenCV build issue, please refer to this MindMap of a setup that should work:

![image](https://user-images.githubusercontent.com/40023722/222990022-b5f4653d-94dd-423c-a235-dcface4ab2ce.png)

Good luck! :rocket:
